configure

```diff
@@ -3631,14 +3638,8 @@ cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 
 /* Override any GCC internal prototype to avoid an error.
    Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.
-   The 'extern "C"' is for builds by C++ compilers;
-   although this is not generally supported in C code supporting it here
-   has little cost and some practical benefit (sr 110532).  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char __gmpz_init (void);
+   builtin and then its argument prototype would still apply.  */
+char __gmpz_init ();
```

Comment:
Why is this changing with this PR?

Reply:
These changes are annoying and caused by different `autoconf` versions.
Each `autoconf` version generates a slightly different `configure` file, but all of them do more or less the same.

The `configure` file that we have in the repo was generated by "Generated by GNU Autoconf 2.71". The PR author used "autoconf 2.71-4".

Since the differences are small, I'm fine with the regenerated `configure` file (even if it contains changes that are not needed).